plugins/signals/src/plugin.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  createApiFactory,
+  createPlugin,
+  discoveryApiRef,
+  identityApiRef,
+} from '@backstage/core-plugin-api';
+import { signalApiRef } from '@backstage/plugin-signals-react';
+import { SignalClient } from './api/SignalClient';
+
+/** @public */
+export const signalsPlugin = createPlugin({
+  id: 'signals',
+  apis: [
+    createApiFactory({
+      api: signalApiRef,
+      deps: {
+        identity: identityApiRef,
+        discoveryApi: discoveryApiRef,
+      },
+      factory: ({ identity, discoveryApi }) =>
+        SignalClient.create({
```

Comment:
Somehow I feel like this should be Singleton instance from the factory so that everyone calling the `useApi(signalsApiRef)` would get the same connection to backend..